contracts/its-stacks-translator/src/contract/its_hub_message_to_clarity_bytes.rs

```diff
@@ -0,0 +1,342 @@
+            ClarityName::from(CLARITY_NAME_SOURCE_CHAIN),
+            Value::string_ascii_from_bytes(source_chain.to_string().into_bytes())
+                .map_err(|_| ContractError::InvalidPayload)?,
+        ),
+        (
+            ClarityName::from(CLARITY_NAME_TOKEN_ID),
+            Value::buff_from(token_id.to_vec()).map_err(|_| ContractError::InvalidPayload)?,
+        ),
+        (
+            ClarityName::from(CLARITY_NAME_NAME),
+            Value::string_ascii_from_bytes(name.into_bytes())
+                .map_err(|_| ContractError::InvalidPayload)?,
+        ),
+        (
+            ClarityName::from(CLARITY_NAME_SYMBOL),
+            Value::string_ascii_from_bytes(symbol.into_bytes())
+                .map_err(|_| ContractError::InvalidPayload)?,
+        ),
+        (
+            ClarityName::from(CLARITY_NAME_DECIMALS),
+            Value::UInt(decimals.into()),
+        ),
+        (
+            ClarityName::from(CLARITY_NAME_MINTER_BYTES),
+            Value::buff_from(if let Some(minter) = minter {
+                minter.to_vec()
+            } else {
+                vec![]
+            })
+            .map_err(|_| ContractError::InvalidPayload)?,
+        ),
+    ])?;
+
+    Ok(Value::from(tuple_data).serialize_to_vec()?)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use axelar_wasm_std::nonempty;
+    use clarity::vm::representations::ClarityName;
+    use clarity::vm::types::{TupleData, Value};
+    use cosmwasm_std::HexBinary;
+    use interchain_token_service_std::{
+        DeployInterchainToken, HubMessage, InterchainTransfer, Message, TokenId,
+    };
+    use router_api::ChainNameRaw;
+
+    use crate::contract::its_hub_message_to_clarity_bytes::{
+        its_hub_message_to_clarity_bytes, MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN,
+        MESSAGE_TYPE_INTERCHAIN_TRANSFER,
+    };
+    use crate::error::ContractError;
+
+    #[test]
+    fn test_its_hub_message_to_clarity_bytes_error() {
+        let token_id: [u8; 32] =
+            from_hex("753306c46380848b5189cd9db90107b15d25decccd93dcb175c0098958f18b6f")
+                .to_vec()
+                .try_into()
+                .unwrap();
+
+        let its_hub_message = HubMessage::SendToHub {
+            destination_chain: ChainNameRaw::from_str("chain").unwrap(),
+            message: Message::DeployInterchainToken(DeployInterchainToken {
+                token_id: TokenId::new(token_id),
+                name: "name".to_string().try_into().unwrap(),
+                symbol: "symbol".to_string().try_into().unwrap(),
+                decimals: 18,
+                minter: None,
+            }),
+        };
+
+        let res = its_hub_message_to_clarity_bytes(its_hub_message);
+
+        assert!(res.is_err());
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            ContractError::InvalidPayload.to_string()
+        );
+    }
+
+    #[test]
+    fn test_its_hub_message_to_clarity_bytes_interchain_transfer() {
```

Comment:
I recommend adding similar tests as abi translator where we use `goldie` to create a golden test for the encoded bytes to ensure those never change accidentally

Reply:
Done